README.md

```diff
@@ -0,0 +1,31 @@
+# Nginx Basic Auth Sidecar
+
+Uses htpasswd.
+
+## Build
+
+```
+$ docker build -t quay.io/app-sre/nginx-gate .
+```
+
+## Run
+
+Given the user `foo` and the password `bar`, there are two ways to inject the
+credentials:
+
+ - Give the container the htpasswd content:
+
+```
+$ htpasswd -bn foo bar
+foo:$apr1$00xPWwOg$sQRfZdrPbGiYz/qUPA8hX1
+$ docker run --rm -it -e HTPASSWD='foo:$apr1$00xPWwOg$sQRfZdrPbGiYz/qUPA8hX1' -e FORWARD_HOST=example.com -p 8080:8080 ng
+```
+
+ - Give the container the username and password:
+
+```
+$ docker run --rm -it -e BASIC_AUTH_USERNAME='foo' -e BASIC_AUTH_PASSWORD='bar' -e FORWARD_HOST=example.com -p 8080:8080 ng
```

Comment:
I'm not sure I like the idea of having the password in plain text in an env var... if an attacker could gain access to dumping the env, or to access an env var, they could exploit this.

What's the use case for this vs simply using `HTPASSWD`?

Reply:
How is that different when compared to all the secrets we expose via Pods environment variables in our clusters?
What I want to achieve here is to have both the plain text password (for client access) and the ecripted on (for nginx configuration) in vault.